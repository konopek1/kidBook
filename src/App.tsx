import React, { ReactNode } from 'react';
import { Box } from '@material-ui/core';
import BottomNavBar from './components/BottomNavBar';
import MainFrame from './components/MainFrame';

export default class App extends React.PureComponent {
  render(): ReactNode {
    return (
      <Box display="flex" flexWrap="wrap" flexDirection="row" height="100vh">
        <MainFrame />
        <BottomNavBar />
      </Box>
    );
  }
}
