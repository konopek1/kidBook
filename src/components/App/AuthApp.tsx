import React from 'react';
import { Box } from '@material-ui/core';
import BottomNavBar from '../navigation/BottomNavBar';
import MainFrame from '../navigation/MainFrame';

const AuthApp: React.FC = () => (
  <Box display="flex" flexDirection="column" justifyContent="space-between" height="100vh">
    <Box overflow="scroll" height="100%"><MainFrame /></Box>
    <BottomNavBar />
  </Box>
);

export default AuthApp;
