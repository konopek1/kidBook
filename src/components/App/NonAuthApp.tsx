import { Box } from '@material-ui/core';
import React from 'react';
import MainFrameNotAuthenticated from '../navigation/MainFrameNotAuthenticated';

const NonAuthApp: React.FC = () => (
  <Box display="flex" flexWrap="wrap" flexDirection="row" height="100vh">
    <MainFrameNotAuthenticated />
  </Box>
);

export default NonAuthApp;
