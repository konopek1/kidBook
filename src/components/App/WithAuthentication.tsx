import { inject } from 'mobx-react';
import React from 'react';
import { compose } from 'recompose';
import Firebase, { withFirebase } from '../../Firebase';
import SessionStore from '../../stores/SessionStore';

export type WithAuthenticationInicjetedProps = {
    sessionStore: SessionStore;
    firebase: Firebase;
};

const withAuthentication = <P extends object>(Component: React.ComponentType<P & WithAuthenticationInicjetedProps>): React.ComponentType<any & WithAuthenticationInicjetedProps> => {
  class WithAuthentication extends React.Component<P & WithAuthenticationInicjetedProps> {
    private listener: any;

    constructor(props: P & WithAuthenticationInicjetedProps) {
      super(props);

      const savedUserJson = localStorage.getItem('authUser');
      const savedUser = savedUserJson !== null ? JSON.parse(savedUserJson) : null;
      this.props.sessionStore.setAuthUser(savedUser);
    }

    componentDidMount() {
      this.listener = this.props.firebase.auth.onAuthStateChanged(
        (authUser: firebase.User | null) => {
          localStorage.setItem('authUser', JSON.stringify(authUser));
          this.props.sessionStore.setAuthUser(authUser);
        },
        () => {
          localStorage.removeItem('authUser');
          this.props.sessionStore.setAuthUser(null);
        },
      );
    }

    componentWillUnmount() {
      this.listener();
    }

    render() {
      return <Component {...this.props} />;
    }
  }

  return compose<any, any & WithAuthenticationInicjetedProps>(
    withFirebase,
    inject('sessionStore'),
  )(WithAuthentication);
};

export default withAuthentication;
