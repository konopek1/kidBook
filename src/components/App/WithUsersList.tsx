import firebase from 'firebase';
import { inject } from 'mobx-react';
import React from 'react';
import { compose } from 'recompose';
import Firebase, { withFirebase } from '../../Firebase';
import { User, userStoreInjected } from '../../stores/UsersStore';

export type WithUsersInicjetedProps = {
    firebase: Firebase;
} & userStoreInjected;

const withUsersList = <P extends object>(Component: React.ComponentType<P & WithUsersInicjetedProps>): React.ComponentType<any & WithUsersInicjetedProps> => {
  class WithUsersList extends React.Component<P & WithUsersInicjetedProps> {
    private listener: any;

    componentDidMount() {
      this.listener = this.props.firebase.db.users().on('value', (userData: firebase.database.DataSnapshot) => {
        if (userData) {
          const mapUsersFromObjectToList = ([uid, user]: [string, Omit<User, 'uid'>]): User => ({ ...user, uid });
          const users = (Object.entries(userData.val()) as [string, Omit<User, 'uid'>][]).map(mapUsersFromObjectToList);
          this.props.userStore.setUsers(users);
        }
      });
    }

    componentWillUnmount() {
      this.listener();
    }

    render() {
      return <Component {...this.props} />;
    }
  }

  return compose<any, any & WithUsersInicjetedProps>(
    withFirebase,
    inject('userStore'),
  )(WithUsersList);
};

export default withUsersList;
