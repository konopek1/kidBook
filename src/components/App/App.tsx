import React from 'react';
import { compose } from 'recompose';
import { observer } from 'mobx-react';
import withAuthentication, { WithAuthenticationInicjetedProps } from './WithAuthentication';
import AuthApp from './AuthApp';
import NonAuthApp from './NonAuthApp';

const App: React.FC<WithAuthenticationInicjetedProps> = (props: WithAuthenticationInicjetedProps) => {
  const { isLoggedIn } = props.sessionStore;
  const component = isLoggedIn ? <AuthApp /> : <NonAuthApp />;
  return component;
};

export default compose<WithAuthenticationInicjetedProps, any>(
  withAuthentication,
  observer,
)(App);
