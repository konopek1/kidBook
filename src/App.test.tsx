import { render } from '@testing-library/react';
import App from './App';
import React from 'react';
import renderer from 'react-test-renderer';

describe('App', () => {

  test('Render properly check snapshot state', () => {
    const appComponent = renderer.create(<App />);
    
    let tree = appComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });

});
