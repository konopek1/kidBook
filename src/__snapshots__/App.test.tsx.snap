// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render properly check snapshot state 1`] = `null`;
